subcommand/controller/command.go

```diff
@@ -111,18 +115,22 @@ func (c *Command) Run(args []string) int {
 		c.UI.Error(fmt.Sprintf("Error parsing -log-level %q: %s", c.flagLogLevel, err.Error()))
 		return 1
 	}
-	// We set UseDevMode to true because we don't want our logs json
-	// formatted.
-	logger := zap.New(zap.UseDevMode(true), zap.Level(zapLevel))
-	ctrl.SetLogger(logger)
-	klog.SetLogger(logger)
+
+	var zapLogger logr.Logger
+	if c.flagLogOutputJSON {
+		zapLogger = zap.New(zap.UseDevMode(true), zap.Level(zapLevel), zap.JSONEncoder())
```

Comment:
I could also do `zap.UseDevMode(false)` if we want to include that in this PR since we intended to do that anyways, or I could submit another PR.